### PR TITLE
Add pre-commit hooks and fix lint on all required files to pass the tests

### PR DESCRIPTION
**Pull request by morenod**
First PR for adding CI on Kube-burner

On this PR, Im adding the pre-commit hooks and I have needed to fix all the lints to pass the test.

We can disable any of them, I have just removed the line length for the markdown files.

### CODE REVIEW

File: test/run-ocp.sh [hunk 1]
```diff
@@ -1,46 +1,49 @@
...
 
 UUID=$(uuidgen)
 ES_SERVER="https://search-perfscale-dev-chmf5l4sh66lvxbnadi4bznl3a.us-west-2.es.amazonaws.com/"
 ES_INDEX="kube-burner-ocp"
-COMMON_FLAGS="--es-server=${ES_SERVER} --es-index=${ES_INDEX} --alerting=true --uuid=${UUID} --qps=5 --burst=5"
+COMMON_FLAGS=" --es-server ${ES_SERVER} --es-index ${ES_INDEX} --alerting=true --uuid=${UUID} --qps=5 --burst=5"
 
 echo "Running node-density wrapper"
-kube-burner ocp node-density --pods-per-node=75 --pod-ready-threshold=10s --container-image=gcr.io/google_containers/pause:3.0 ${COMMON_FLAGS}
+kube-burner ocp node-density --iterations=2 --pods-per-node=75 --pod-ready-threshold=10s --container-image=gcr.io/google_containers/pause:3.0 "${COMMON_FLAGS}"
```
Comment (rsevilla87) [inline test/run-ocp.sh:24]: The iterations flag shouldn't be here, this is breaking the tests

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  - repo: https://github.com/golangci/golangci-lint
+    rev: v1.52.2
+    hooks:
+      - id: golangci-lint
+        args: [--timeout=5m]
+  - repo: https://github.com/markdownlint/markdownlint
+    rev: v0.12.0
+    hooks:
+      - id: markdownlint
+        args: [-r, ~MD013] # MD013 Line length
```
Comment (vishnuchalla) [inline .pre-commit-config.yaml:11]: Just curious, why did we decide on ignoring the [line-length](https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md013---line-length) , deosn't it help us in maintaining consistency throughout our documentation?

**Comment by vishnuchalla** [comment]
As we also have some YAMl files in the repo, do we want to consider a hook for yaml linting as well : https://github.com/adrienverge/yamllint ?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (morenod) [reply .pre-commit-config.yaml:11]: imho, line lenght is obsolete. with current monitors, 80chars do not help to read the code

File: test/run-ocp.sh (hunk 1 above)
Comment (morenod) [reply test/run-ocp.sh]: Fixed

**Comment by morenod** [comment]
> As we also have some YAMl files in the repo, do we want to consider a hook for yaml linting as well : https://github.com/adrienverge/yamllint ?

I have tried to add a yaml linter but most of the files we have in the repo are yaml templates named as yaml files, containing variables.

yaml lint fails on this kind of files when it finds a var. changing this files to yaml.template should resolve the issue, but it will requires changes on the code, which is out of scope on this PR.

If we want to lint yaml files, first we need to move yaml template files to the correct extension

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Comment by vishnuchalla** [comment]
looks good to me too!

**Review by vishnuchalla — APPROVED** [review]

**Pull request merged**